src/app/pages/AccountPage/Features/TransactionHistory/index.tsx

```diff
@@ -11,39 +11,95 @@ import { useTranslation } from 'react-i18next'
 import { useSelector } from 'react-redux'
 
 import {
+  hasAccountUnknownPendingTransactions,
   selectAccountAddress,
+  selectPendingTransactionForAccount,
   selectTransactions,
   selectTransactionsError,
 } from 'app/state/account/selectors'
 import { selectSelectedNetwork } from 'app/state/network/selectors'
 import { ErrorFormatter } from 'app/components/ErrorFormatter'
-
-interface Props {}
+import { AlertBox } from '../../../../components/AlertBox'
+import { Anchor } from 'grommet/es6/components/Anchor'
+import { Text } from 'grommet/es6/components/Text'
+import { FormNext } from 'grommet-icons/es6/icons/FormNext'
+import { config } from '../../../../../config'
+import { backend } from '../../../../../vendors/backend'
 
 /**
  * Displays the past transactions from the state for a given account
  */
-export function TransactionHistory(props: Props) {
+export function TransactionHistory() {
   const { t } = useTranslation()
   const allTransactions = useSelector(selectTransactions)
   const transactionsError = useSelector(selectTransactionsError)
   const address = useSelector(selectAccountAddress)
+  const pendingTransactions = useSelector(selectPendingTransactionForAccount)
+  const hasUnknownPendingTransactions = useSelector(hasAccountUnknownPendingTransactions)
   const network = useSelector(selectSelectedNetwork)
-  const transactionComponents = allTransactions.map((t, i) => (
-    <Transaction key={i} transaction={t} referenceAddress={address} network={network} />
+  const backendLinks = config[network][backend()]
+  const transactionComponents = allTransactions.map(t => (
+    <Transaction key={t.hash} transaction={t} referenceAddress={address} network={network} />
   ))
+  const pendingTransactionComponents = pendingTransactions
+    .filter(({ hash: pendingTxHash }) => !allTransactions.some(({ hash }) => hash === pendingTxHash))
+    .map(t => <Transaction key={t.hash} transaction={t} referenceAddress={address} network={network} />)
 
   return (
-    <Box gap="medium" margin="none">
+    <Box margin="none">
       {transactionsError && (
         <p>
-          {t('account.transaction.loadingError', "Couldn't load transactions.")}{' '}
+          {t('account.transaction.loadingError', `Couldn't load transactions.`)}{' '}
           <ErrorFormatter code={transactionsError.code} message={transactionsError.message} />
         </p>
       )}
+      {/* eslint-disable no-restricted-syntax */}
+      {(!!pendingTransactionComponents.length || hasUnknownPendingTransactions) && (
```

Comment:
It shows pending transactions while loading. I would default to not showing them - it's much more common.

[screen-recording.webm](https://github.com/oasisprotocol/oasis-wallet-web/assets/3758846/eb169067-4b7e-49a2-b2c5-50072159ddd4)